File: servers/dotnet/Domain/Blogs/Blog.cs
using Domain.Users;

namespace Domain.Blogs;

public class Blog
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required string Title { get; set; }
    public required string Body { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public required string UserId { get; set; }
    public User User { get; set; } = null!;

}

File: clients/react/src/lib/types/Blog.ts
import { BlogUserDto } from './User';

export type BlogDto = {
  id: string;
  title: string;
  body: string;
  createdAt: Date;
  user: BlogUserDto;
};

export type CreateBlogDto = {
  title: string;
  body: string;
};

export type BlogSearchCriteria = {
  userId?: Array<string>;
};
